syntax = "proto3";

package snsFSynch;

import "google/protobuf/timestamp.proto";

service SNSFSynch{
    rpc Contact (Message) returns (Reply) {}
    // follow function to add user to following list
    rpc Follow (Request) returns (Reply) {}
    // timeline modifier
    rpc Timeline (TimelineMsg) returns (Reply) {}
    // add user to all user list
    rpc NotifyFailure (Message) returns (Reply) {}
    // get the all user list from synchronizer
    rpc AllUsers (Request) returns (userReply) {}
}   

// initial commuincation
message Message{
    // fsynch id 
    int32 id = 1;
    // server info host:port
    string server_info = 2;
}

// async request between synchronizer
message Request{
    int32 follower = 1; // user following
    int32 followed = 2; // user being followed
}

message Reply{
    string msg = 1;
}

// list of all the message in the timeline user has sent
message TimelineMsg{
    int32 id = 1;
    string post = 2;
}

message userReply{
    repeated string users = 1;
}